tkg/client/cluster.go

```diff
@@ -160,7 +160,13 @@ func (c *TkgClient) CreateCluster(options *CreateClusterOptions, waitForCluster
 			return false, errors.Wrap(err, "unable to get cluster configuration")
 		}
 
-		if !c.IsFeatureActivated(constants.FeatureFlagAllowLegacyCluster) {
+		allowLegacyCluster, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAllowLegacyCluster)
+		if err != nil {
+			log.Error(err, "ALLOW_LEGACY_CLUSTER variable is set failed")
```

Comment:
What does the variable here mean?

Reply:
@wjun 
>I wonder if the error means 'variable is not set' actually?

when get api has some issues will also cause error, so can I modify the error message to `Get ALLOW_LEGACY_CLUSTER variable's value failed`?

Reply:
Is err not nil when ALLOW_LEGACY_CLUSTER is absent? Or just it raise an error only when type is not bool?

Reply:
@DanielXiao , when err is not nil, that means ALLOW_LEGACY_CLUSTER is absent or get api has some issues. But actually, in this place, if error happens, that only means get api has issues. Because before this check, we have set `ALLOW_LEGACY_CLUSTER` [here](https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/client/common.go#L367). [SetAllowLegacyClusterConfiguration](https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/client/common.go#L310-L342)  func has checked if ALLOW_LEGACY_CLUSTER is absent or not. If it's absent, then we will set it.

Reply:
**suggestion:** Why don't we use something like the following as the message?

```go
fmt.Sprintf("failed getting environment variable %s", constants.ConfigVariableAllowLegacyCluster)
```